state/event_table_test.go

```diff
@@ -1127,3 +1127,90 @@ func TestEventTableRedactMissingOK(t *testing.T) {
 			}`),
 		}}))
 }
+
+func TestEventTable_SelectLatestEventsBetween_MissingPrevious(t *testing.T) {
+	db, close := connectToDB(t)
+	defer close()
+	table := NewEventTable(db)
+	txn, err := db.Beginx()
+	if err != nil {
+		t.Fatalf("failed to start txn: %s", err)
+	}
+	defer txn.Rollback()
+	roomID := fmt.Sprintf("!%s", t.Name())
+	events := []Event{
+		{ID: "A", MissingPrevious: false},
+		{ID: "B", MissingPrevious: false},
+		{ID: "C", MissingPrevious: false},
+		{ID: "D", MissingPrevious: true},
```

Comment:
Be kinder and mux in "missing previous" to the ID please, else I need to remember it's D,G,H.